prow/userdashboard/userdashboard.go

```diff
@@ -168,24 +195,50 @@ func (da *DashboardAgent) HandleUserDashboard(queryHandler PullRequestQueryHandl
 
 // Query function that returns a list of open pull requests owned by the user whose access token
 // is consumed by the github client.
-func (da *DashboardAgent) Query(ctx context.Context, ghc githubClient) ([]PullRequest, error) {
-	var prs = []PullRequest{}
+func (da *DashboardAgent) Query(ctx context.Context, ghc githubClient, query string) ([]PullRequest, error) {
+	var prs []PullRequest
 	vars := map[string]interface{}{
-		"prsCursor": (*githubql.String)(nil),
+		"query":        (githubql.String)(query),
+		"searchCursor": (*githubql.String)(nil),
 	}
+	var totalCost int
+	var remaining int
 	for {
-		pq := PullRequestQuery{}
-		if err := ghc.Query(ctx, &pq, vars); err != nil {
+		sq := searchQuery{}
+		if err := ghc.Query(ctx, &sq, vars); err != nil {
 			return nil, err
 		}
-		for _, n := range pq.Viewer.PullRequests.Nodes {
+		totalCost += int(sq.RateLimit.Cost)
+		remaining = int(sq.RateLimit.Remaining)
+		for _, n := range sq.Search.Nodes {
 			prs = append(prs, n.PullRequest)
 		}
-		if !pq.Viewer.PullRequests.PageInfo.HasNextPage {
+		if !sq.Search.PageInfo.HasNextPage {
 			break
 		}
-		vars["prsCursor"] = githubql.NewString(pq.Viewer.PullRequests.PageInfo.EndCursor)
+		vars["searchCursor"] = githubql.NewString(sq.Search.PageInfo.EndCursor)
 	}
-
+	da.log.Infof("Search for query \"%s\" cost %d point(s). %d remaining.", query, totalCost, remaining)
 	return prs, nil
 }
+
+func (da *DashboardAgent) ConstructSearchQuery(login string) string {
+	tokens := []string{"is:pr", "state:open", "author:" + login}
+	for i := range da.repos {
+		tokens = append(tokens, fmt.Sprintf("repo:\"%s\"", da.repos[i]))
```

Comment:
I'm not sure how Github will behave with long search queries or queries that specify large numbers of repos. We should test that the API can handle queries like this.
We could filter out PRs from these repos ourselves instead of specifying the repos in the query if needed.

Reply:
Does query token cost increase with complexity? That would be another motivation to do the filtering client-side here as well

Reply:
I found no information about how many repo tokens allowed. Probably we should query all prs and filter outs by ourself.

Reply:
token cost is related to nodes touched in graphQL. you can write your query to return the cost of it if you want to debug this.